express-crud/server.ts

```diff
@@ -0,0 +1,145 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const { gradeId } = req.params;
+    if (!Number(gradeId)) {
+      throw new ClientError(400, 'gradeId must be a positive integer');
+    }
+    const sql = `
+      select * from "grades"
+      where "gradeId" = $1;
+    `;
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (!grade) throw new ClientError(404, `gradeId ${gradeId} not found`);
+    res.status(200).json(grade);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `select * from "grades";`;
+    const result = await db.query(sql);
+    const grades = result.rows;
+    if (!grades) throw new ClientError(400, 'requested data does not exist');
```

Comment:
This actually won't ever happen because of there are no grades then `result.rows` will be `[]`. In that case, you will just return an empty array with status 200, which is a very reasonable thing to do.